Parse/src/main/java/com/parse/ParseEncoder.java

```diff
@@ -120,6 +122,12 @@ public Object encode(Object object) {
         return ((ParseQuery.RelationConstraint) object).encode(this);
       }
 
+      if (object instanceof Pattern) {
+        JSONObject json = new JSONObject();
+        json.put("$regex", object.toString());
```

Comment:
It makes sense. Also since whereContainsAll is documented to work only with `ParseObject`s, you could call directly `builder.addCondition(key, "$all", startsWithConstraints)` and return this